pkg/trace/remoteconfighandler/remote_config_handler.go

```diff
@@ -29,24 +35,37 @@ type rareSampler interface {
 
 // RemoteConfigHandler holds pointers to samplers that need to be updated when APM remote config changes
 type RemoteConfigHandler struct {
-	remoteClient    config.RemoteClient
-	prioritySampler prioritySampler
-	errorsSampler   errorsSampler
-	rareSampler     rareSampler
-	agentConfig     *config.AgentConfig
+	remoteClient                  config.RemoteClient
+	prioritySampler               prioritySampler
+	errorsSampler                 errorsSampler
+	rareSampler                   rareSampler
+	agentConfig                   *config.AgentConfig
+	configState                   *state.AgentConfigState
+	configSetEndpointFormatString string
 }
 
 func New(conf *config.AgentConfig, prioritySampler prioritySampler, rareSampler rareSampler, errorsSampler errorsSampler) *RemoteConfigHandler {
 	if conf.RemoteSamplingClient == nil {
 		return nil
 	}
```

Comment:
Should `RemoteSamplingClient` be renamed to something more generic e.g `RemoteClient`? If I understand correctly the same client will be used for sampling **and** for log level config from now on.